### PR TITLE
Acceptance test to restore a file from trashbin whose parent folder is renamed

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance to restore a file from trashbin whose parent folder is renamed

## Related Issue

-Related to issue : https://github.com/owncloud/phoenix/issues/1753

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6623/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6661/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6667/

<p>





































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Resharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6664/

<p>

























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6664/

<p>























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6667/

<p>























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6664/

<p>







































































































</p></details>

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 1]
```diff
@@ -538,6 +538,13 @@ module.exports = {
         .click(formattedFileRow)
         .waitForElementVisible('@versionsElement')
         .click('@versionsElement')
+    },
+    attemptToRestoreFile: function (fileName) {
+      return this.initAjaxCounters()
+        .waitForFileWithPathVisible(fileName)
+        .useXpath()
+        .performFileAction(fileName, 'restore')
+        .waitForElementVisible('@restorationAlert')
```
Comment (HariBhandari07) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: attach useCss() at the end

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature [hunk 2]
```diff
@@ -97,3 +97,20 @@ Feature: Restore deleted files/folders
...
+  Scenario: Restore a file from trashbin whose parent folder is renamed
+    Given the user has created file "simple-folder/file-to-delete-and-restore"
+    And the following files have been deleted by user "user1"
+      | name                                     |
+      | simple-folder/file-to-delete-and-restore |
+    And user "user1" has renamed folder "simple-folder" to "simple-folder-renamed"
+    When the user browses to the trashbin page
+    #And the user restores file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    And the user tries to restore file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    #Then as "user1" the file with original path "simple-folder/file-to-delete-and-restore" should not exist in trash
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: Error message is also displayed in the webui, add the step describing that.

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 3]
```diff
@@ -538,6 +538,13 @@ module.exports = {
         .click(formattedFileRow)
         .waitForElementVisible('@versionsElement')
         .click('@versionsElement')
+    },
+    attemptToRestoreFile: function (fileName) {
+      return this.initAjaxCounters()
+        .waitForFileWithPathVisible(fileName)
+        .useXpath()
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: Is this necessary here?

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: done

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: removed

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature [hunk 4]
```diff
@@ -97,3 +97,22 @@ Feature: Restore deleted files/folders
...
+    Given the user has created file "simple-folder/file-to-delete-and-restore"
+    And the following files have been deleted by user "user1"
+      | name                                     |
+      | simple-folder/file-to-delete-and-restore |
+    And user "user1" has renamed folder "simple-folder" to "simple-folder-renamed"
+    When the user browses to the trashbin page
+    #And the user restores file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    And the user tries to restore file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    Then an error with message "Restoration of file-to-delete-and-restore failed" should be displayed
+    #Then a success message "" should be displayed
```
Comment (dpakach) [inline tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: This step is a bit confusing. Please add correct error message or remove the step.

Comment (jasson99) [reply tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: done

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature [hunk 5]
```diff
@@ -97,3 +97,22 @@ Feature: Restore deleted files/folders
...
+    When the user browses to the trashbin page
+    #And the user restores file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    And the user tries to restore file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    Then an error with message "Restoration of file-to-delete-and-restore failed" should be displayed
+    #Then a success message "file-to-delete-and-restore was restored successfully" should be displayed
+    #Then as "user1" the file with original path "simple-folder/file-to-delete-and-restore" should not exist in trash
+    And as "user1" the file with original path "simple-folder/file-to-delete-and-restore" should exist in trash
+    When the user browses to the folder "simple-folder-renamed" on the files page
+    #Then file "file-to-delete-and-restore" should be listed on the webUI
+    Then file "file-to-delete-and-restore" should not be listed on the webUI
```
Comment (skshetry) [inline tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: I think, this is redundant here.

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature [hunk 6]
```diff
@@ -97,3 +97,22 @@ Feature: Restore deleted files/folders
...
+  Scenario: Restore a file from trashbin whose parent folder is renamed
+    Given the user has created file "simple-folder/file-to-delete-and-restore"
+    And the following files have been deleted by user "user1"
+      | name                                     |
+      | simple-folder/file-to-delete-and-restore |
+    And user "user1" has renamed folder "simple-folder" to "simple-folder-renamed"
+    When the user browses to the trashbin page
+    #And the user restores file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    And the user tries to restore file "simple-folder/file-to-delete-and-restore" from the trashbin using the webUI
+    Then an error with message "Restoration of file-to-delete-and-restore failed" should be displayed
```
Comment (skshetry) [inline tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: There's already a step:

```suggestion
    Then the error message "Restoration of file-to-delete-and-restore failed" should be displayed on the webUI
```

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 7]
```diff
@@ -539,6 +540,20 @@ module.exports = {
...
+        .performFileAction(fileName, 'restore')
+        .waitForElementVisible('@restorationAlert')
+    },
+    failToRestore: function (errorMessage) {
+      const requiredXpath = this.elements.restorationAlert.selector +
+                            util.format(this.elements.restorationError.selector, errorMessage)
+      return this
+        .useXpath()
+        .waitForElementVisible(requiredXpath)
+        .useCss()
```
Comment (skshetry) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: No need for this command.

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature (hunk 5 above)
Comment (jasson99) [reply tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: I think its fine here

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js (hunk 7 above)
Comment (jasson99) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: done

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature (hunk 6 above)
Comment (jasson99) [reply tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: done

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature (hunk 5 above)
Comment (jasson99) [reply tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: done

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6840/

<p>








</p></details>

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 8]
```diff
@@ -175,10 +175,10 @@ module.exports = {
      */
     restoreFile: function (fileName) {
       return this.initAjaxCounters()
-        .waitForFileVisible(fileName)
+        .waitForFileWithPathVisible(fileName)
         .useXpath()
         .performFileAction(fileName, 'restore')
-        .waitForElementNotPresent(this.getFileRowSelectorByFileName(fileName))
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: I think we should add an `waitForOutstandingAjaxCalls`

Comment (jasson99) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: done

**Pull request merged**